intake_netcdf/__init__.py

```diff
@@ -70,17 +77,17 @@ def _get_schema(self):
 
     def read(self):
         self._load_metadata()
-        return self._ds
+        return self._ds.load()
 
     def read_chunked(self):
-        raise Exception('read_chunked not supported for xarray containers.')
+        self._load_metadata()
+        return self._ds
```

Comment:
Yeah, so this is approximately correct, but it doesn't give something you can iterate over. I'm not sure whether this should be an exception just like get_partition.